packages/template-retail-react-app/app/pages/product-list/index.jsx

```diff
@@ -543,7 +544,8 @@ const ProductList = (props) => {
                                 spacingX={4}
                                 spacingY={{base: 12, lg: 16}}
                             >
-                                {isHydrated() && (isRefetching || !productSearchResult)
+                                {isHydrated() &&
+                                ((isRefetching && !isFetched) || !productSearchResult)
```

Comment:
Would you mind adding test case for `isFetched` && `isRefetching` and `!isFetched && isRefetching`?